Python/Product/Analysis/ProjectEntry.cs

```diff
@@ -280,60 +283,31 @@ where fileName.StartsWith(pathPrefix)
             MyScope.ClearUnresolvedModules();
         }
 
-        public IGroupableAnalysisProject AnalysisGroup {
-            get {
-                return _projectState;
-            }
-        }
+        public IGroupableAnalysisProject AnalysisGroup => ProjectState;
 
-        public ModuleAnalysis Analysis {
-            get { return _currentAnalysis; }
-        }
+        public ModuleAnalysis Analysis { get; private set; }
 
-        public string FilePath {
-            get { return _filePath; }
-        }
+        public string FilePath { get; }
 
-        public IAnalysisCookie Cookie {
-            get { return _cookie; }
-        }
+        public IAnalysisCookie Cookie { get; private set; }
 
-        internal PythonAnalyzer ProjectState {
-            get { return _projectState; }
-        }
+        internal PythonAnalyzer ProjectState { get; private set; }
 
-        public PythonAst Tree {
-            get { return _tree; }
-        }
+        public PythonAst Tree { get; private set; }
```

Comment:
This property is accessed under lock inside `ProjectEntry` but is available without lock `ModuleAnalysis` and several other places

Reply:
Reading Tree doesn't require the lock - Cookie and _updatesPending do. GetTreeAndCookie reads under the lock to ensure you get matched tree and cookie, but directly accessing tree is not guaranteed to be the latest version.